### PR TITLE
Update to allow cancel pass case to be sent to COMET

### PR DESCRIPTION
**Pull request by SteveScorfield**

### CODE REVIEW

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/config/GatewayEventsConfig.java [hunk 1]
```diff
@@ -10,13 +10,14 @@
...
   public static final String CANONICAL_CANCEL_SENT = "Canonical - Action Cancel Sent";
+  public static final String CANONICAL_CANCEL_FAILED ="Canonical - Action cancel failed. Address type is not a Household (HH)";
   public static final String RM_REQUEST_RECEIVED = "RM - Request Received";
 
   @Bean
   public GatewayEventManager gatewayEventManager() {
     GatewayEventManager gatewayEventManager = new GatewayEventManager();
     gatewayEventManager.addEventTypes(new String[] {CANONICAL_CREATE_SENT, CANONICAL_UPDATE_SENT, CANONICAL_CANCEL_SENT,
-        RM_REQUEST_RECEIVED});
+        RM_REQUEST_RECEIVED, CANONICAL_CANCEL_FAILED});
```
Comment (JacobHarrisonONS) [inline src/main/java/uk/gov/ons/census/fwmt/rmadapter/config/GatewayEventsConfig.java:20]: this canonical cancel failed, does it work as you intended it to? You'll have to explain how you are using it within the event logger.

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java [hunk 2]
```diff
@@ -31,9 +30,13 @@ public void sendJobRequest(ActionInstruction actionInstruction) throws GatewayEx
...
     } else if (actionInstruction.getActionCancel() != null) {
-      jobServiceProducer.sendMessage(CanonicalJobHelper.newCancelJob(actionInstruction));
-      gatewayEventManager
-          .triggerEvent(actionInstruction.getActionCancel().getCaseId(), CANONICAL_CANCEL_SENT, LocalTime.now());
+      if (actionInstruction.getActionCancel().getAddressType().equals("HH")) {
+        jobServiceProducer.sendMessage(CanonicalJobHelper.newCancelJob(actionInstruction));
+        gatewayEventManager
+            .triggerEvent(actionInstruction.getActionCancel().getCaseId(), CANONICAL_CANCEL_SENT, LocalTime.now());
+      } else {
+        gatewayEventManager.triggerEvent(actionInstruction.getActionCancel().getCaseId(), CANONICAL_CANCEL_FAILED, LocalTime.now());
```
Comment (JacobHarrisonONS) [inline src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java:38]: ah i see you are sending error messages when your address type isn't HH. i don't think this is required. May be able to just ignore and dead letter the message instead?

File: src/main/resources/xsd/actionInstruction.xsd [hunk 3]
```diff
@@ -159,6 +159,7 @@
                     <xs:element name="reason" type="xs:string"/>
                     <xs:element name="caseId" type="xs:string" minOccurs="0" maxOccurs="1"/>
                     <xs:element name="caseRef" type="xs:string" minOccurs="0" maxOccurs="1"/>
+                    <xs:element name="addressType" type="xs:string" minOccurs="0" maxOccurs="1"/>
```
Comment (JacobHarrisonONS) [inline src/main/resources/xsd/actionInstruction.xsd:162]: has this change been accepted by Dave Richards etc?

**Review by JacobHarrisonONS — CHANGES_REQUESTED** [review]

File: src/main/resources/xsd/actionInstruction.xsd (hunk 3 above)
Comment (SteveScorfield) [reply src/main/resources/xsd/actionInstruction.xsd:162]: Checked with Olu and it is expected to be here.

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java (hunk 2 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java:38]: Part of the ticket wants an error log to be sent if the case is not HH.

Comment (JacobHarrisonONS) [reply src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java:38]: Okay 👍

File: src/main/resources/xsd/actionInstruction.xsd (hunk 3 above)
Comment (JacobHarrisonONS) [reply src/main/resources/xsd/actionInstruction.xsd:162]: okay 🌮

**Review by JacobHarrisonONS — APPROVED** [review]
Looks good 👍

**Pull request merged**